src/category/category.service.ts

```diff
@@ -37,4 +37,41 @@ export class CategoryService {
     })
     return allCategories
   }
+
+  async getAllCategories(
+    page: number = 1,
+    limit: number = 50,
+    filter: CategoryFilterDto = {},
+  ) {
+    const { name } = filter
+
+    const filterConditions: any = {}
+
+    if (name) {
+      filterConditions.name = { contains: name, mode: 'insensitive' }
```

Comment:
Similar comment as the book service, maybe it would be best to have a shared folder or utils for the filters